File: Universe_Backend/Data/Models/PostReaction.cs
using Universe_Backend.Data.DTOs;

namespace Universe_Backend.Data.Models;

public class PostReaction : Reaction
{
    public int PostId { get; set; }
    public virtual Post? Post { get; set; }

    public PostReactionDTO ToDTO()
    {
        return new PostReactionDTO
        {
            Id = Id,
            UserId = UserId,
            ReactionType = ReactionType,
            ReactionDate = ReactionDate,
            PostId = PostId
        };
    }
}

File: universe/lib/models/data/post_reaction.dart
import 'package:json_annotation/json_annotation.dart';

part 'post_reaction.g.dart';

@JsonSerializable()
class PostReaction {
  final int id;
  final String userId;
  final String reactionType;
  final DateTime reactionDate;
  final int postId;

  PostReaction({
    this.id = -1,
    required this.userId,
    required this.reactionType,
    required this.reactionDate,
    required this.postId,
  });

  factory PostReaction.fromJson(Map<String, dynamic> data) =>
      _$PostReactionFromJson(data);

  Map<String, dynamic> toJson() => _$PostReactionToJson(this);
}
